### PR TITLE
new type of chart added - area-spline with metrics

### PR DESCRIPTION
**Pull request by apryka**
## Related tickets?

https://github.com/keen/keen-dataviz.js/issues/127

## How should this be tested?

Step through the code line by line. Things to keep in mind as you review:
 - Are there any edge cases not covered by this code?
 - Does this code follow conventions (naming, formatting, modularization, etc) where applicable?

Fetch the branch and/or deploy to staging to test the following:

- [ ] Does the code compile without warnings (check shell, console)?
- [ ] Do all tests pass?
- [ ] Does the UI, pixel by pixel, look exactly as expected (check various screen sizes, including mobile)?
- [ ] If the feature makes requests from the browser, inspect them in the Web Inspector. Do they look as expected (parameters, headers, etc)?
- [ ] If the feature sends data to Keen, is the data visible in the project if you run an extraction (include link to collection/query)?
- [ ] If the feature saves data to a database, can you confirm the data is indeed created in the database?

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -795,6 +795,24 @@ Dataviz.version = pkg.version;
...
+  if (percentDifference < 0) {
+    labelClass = 'keen-dataviz-metric-red';
+    iconClass = 'arrow-red';
+  }
+
+  return `
+    <div class="percent-difference ${labelClass}"><div class="${iconClass}"></div> ${percentDifference} %</div>
+    <div class="current-count ${labelClass}">${count}</div>`;
+}
+
```
Comment (adamkasprowicz) [inline lib/index.js]: let's move this code to /lib/libraries/default/metric-combo.js

File: lib/index.js [hunk 2]
```diff
@@ -814,15 +832,22 @@ function buildDomWrapper(el, options) {
...
   if (options.title && options.showTitle) {
     html += `<div class="keen-dataviz-title ${options.theme}-title">${options.title}</div>`;
   }
+  if (options.showMetrics && options.type === 'area-spline') {
+    if (!options.currentQuery || !options.previousQuery) {
+      html += '<div class="no-data">No data to display</div>';
+    } else {
+      html += generateAreaWithMetricsDom(options.previousQuery, options.currentQuery);
+    }
+  }
```
Comment (adamkasprowicz) [inline lib/index.js]: let's not modify buildDomWrapper function

File: lib/libraries/index.js [hunk 3]
```diff
@@ -289,6 +289,44 @@ function defineC3(){
           }
         }
 
+        if (options.showMetrics && type === 'area-spline') {
```
Comment (adamkasprowicz) [inline lib/libraries/index.js]:
```
if (type === 'metric-combo') {
```

File: lib/style/keen-dataviz.css [hunk 4]
```diff
@@ -140,6 +140,29 @@
...
+    right: 0.5em;
+    font-size: 80%;
+  }
+
+  .arrow-green,
+  .arrow-red {
+    left: -1em;
+    border-width: 5px;
+  }
+
```
Comment (adamkasprowicz) [inline lib/style/keen-dataviz.css]: let's encapsulate these classes in something that we'll use just for metric-combo charts

File: test/demo/index.html [hunk 5]
```diff
@@ -303,6 +303,11 @@ <h3>Timeseries (secondly)</h3>
           <div class="chart-c3-area-spline-time-stacked keen-chart-container"></div>
         </div>
       </div>
+      <div class="keen-ui-row">
+        <div class="keen-ui-col-4">
+          <div class="chart-c3-area-spline-metrics keen-chart-container"></div>
```
Comment (adamkasprowicz) [inline test/demo/index.html]: let's use custom width & height - the same as in the auto-collector-dashboard repo:
width: 270px
height: 120px

File: test/demo/index.html [hunk 6]
```diff
@@ -303,6 +303,11 @@ <h3>Timeseries (secondly)</h3>
...
+        </div>
+      </div>
```
Comment (adamkasprowicz) [inline test/demo/index.html]: height of the chart is bigger than the container's height

File: test/demo/index.html [hunk 7]
```diff
@@ -1151,6 +1164,21 @@ <h3>Timeseries</h3>
...
+          previousQuery: areaSplineWithMetricsData.previous,
+          currentQuery: areaSplineWithMetricsData.current,
+          results: areaSplineWithMetricsData.interval,
+          tooltip: {
+          format: {
+            title: (d) => new Date(d).toLocaleString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true })
+            }
+          }
+        });
+
```
Comment (adamkasprowicz) [inline test/demo/index.html]:
```javascript
        window.areaSplineWithMetrics = new KeenDataviz({
          container: '.chart-c3-metric-combo',
          title: 'Metric Combo',
          type: 'metric-combo',
          comboResults: {
            previous: areaSplineWithMetricsData.previous,
            current: areaSplineWithMetricsData.current,
            interval: areaSplineWithMetricsData.interval,
          },

          tooltip: {
            format: {
              title: (d) => new Date(d).toLocaleString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true })
            }
          }
        });
```

File: test/demo/index.html [hunk 8]
```diff
@@ -303,6 +303,11 @@ <h3>Timeseries (secondly)</h3>
           <div class="chart-c3-area-spline-time-stacked keen-chart-container"></div>
         </div>
       </div>
+      <div class="keen-ui-row">
+        <div class="keen-ui-col-4">
+          <div class="chart-metric-combo keen-chart-container"></div>
+        </div>
+      </div>
```
Comment (adamkasprowicz) [inline test/demo/index.html]: at the bottom of the page - create h2 called "combo charts"
let's move this chart there

File: lib/libraries/default/metric-combo.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+        columns.push(this.dataset.selectColumn(i));
+      }
+    });
+
+    const prevResult = (metricResults && metricResults.previous && metricResults.previous.result)
+      || 0;
+    const currResult = (metricResults && metricResults.current && metricResults.current.result)
+      || 0;
+    const count = currResult - prevResult;
+    const percentDifference = prevResult === 0 ? 'Not defined' : Math.round((count / prevResult) * 100);
```
Comment (adamkasprowicz) [inline lib/libraries/default/metric-combo.js]: let's imagine scenario, that I'm counting singups.
Last week I had 0
This week I have 10
If I use this component - I'll see "Not defined' in the percentage change, even if both of my queries return correct results.
Let's use `-` instead of `not defined`

File: test/demo/index.html [hunk 10]
```diff
@@ -1151,6 +1164,23 @@ <h3>Timeseries</h3>
...
+            current: metricComboChartData.current,
+          },
+          results: metricComboChartData.interval,
+          tooltip: {
+            format: {
+              title: (d) => new Date(d).toLocaleString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true })
+            }
+          }
+        })
+        .render();
```
Comment (adamkasprowicz) [inline test/demo/index.html]: please add this chart type to the list
https://github.com/keen/keen-dataviz.js/tree/master/docs#custom-built-by-us
between metric and message

File: docs/README.md [hunk 11]
```diff
@@ -40,6 +40,22 @@ const chart = new KeenDataviz({
   })
 ```
 
+* combo chart
```
Comment (adamkasprowicz) [inline docs/README.md]: metric combo
